pints/_optimisers/_hager_zhang_line_search.py

```diff
@@ -0,0 +1,899 @@
+#
+# Fixed learning-rate gradient descent.
+#
+# This file is part of PINTS (https://github.com/pints-team/pints/) which is
+# released under the BSD 3-clause license. See accompanying LICENSE.md for
+# copyright notice and full license details.
+#
+from __future__ import absolute_import, division
+from __future__ import print_function, unicode_literals
+
+import pints
+import numpy as np
+from numpy.linalg import norm
+
+
+class HagerZhang(pints.Optimiser):
+    """
+    Gradient-descent method with a fixed learning rate.
```

Comment:
Needs a docstring with references etc.

Reply:
Pseudocode too would I think be really useful. For pseudocode in the other methods, we tend not to incorporate ask/tell but rather lay out the algorithm as given in the paper (but made tidy and understandable for us).